core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -148,10 +148,15 @@ class AdminClient(val time: Time,
       throw new IllegalArgumentException(s"Consumer Group $groupId with protocol type '${metadata.protocolType}' is not a valid consumer group")
 
     Errors.forCode(metadata.errorCode()).maybeThrow()
-    val consumers = metadata.members.map { consumer =>
-      val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(Utils.readBytes(consumer.memberAssignment)))
-      ConsumerSummary(consumer.memberId, consumer.clientId, consumer.clientHost, assignment.partitions.toList)
-    }.toList
+    val consumers = metadata.state match {
+      case "PreparingRebalance" | "AwaitingSync" =>
+        List()
+      case _ =>
+        metadata.members.map { consumer =>
+          val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(Utils.readBytes(consumer.memberAssignment)))
```

Comment:
After thinking about this, parsing the assignment really only makes sense if the group is "Stable," so maybe we ought to handle that case specifically rather than as the default. Then for the default case, maybe we could include all members in the group and just use an empty assignment?